### PR TITLE
6X: Fix memory leak issues which reported by Coverity

### PR DESCRIPTION
**Pull request by interma**
These issues are reported by Coverity: [detail link](https://pivotal.polaris.synopsys.com/projects/34158812-801e-47d3-8aad-492774ea4d91/branches/12d9ee7f-fb2e-4153-be92-bc147e3a5603/issues?filter=issue%5Bstatus%5D%5B%24eq%5D%3Dopened%26issue%5Btaxonomy%5D%5Bid%5D%5B4bae258d-865d-4a06-9ff8-a7e141cabb2d%5D%5Btaxon%5D%5B%24eq%5D%3Dhigh%26issue%5Btriage-status%5D%5B%24eq%5D%3Dto-be-fixed)

Actually, more issues are found, but some of them are the same as upstream code 
and not a very big deal (e.g. in utility, like pg_dump), so ignore them.

Just fix issues which gpdb introduced in this PR.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [x] Review a PR in return to support the community

### CODE REVIEW

**Review by QingMa-gp — APPROVED** [review]

File: src/backend/cdb/motion/ic_tcp.c [hunk 1]
```diff
@@ -2232,6 +2232,7 @@ format_sockaddr(struct sockaddr *sa, char *buf, int bufsize)
 	else if (sa->sa_family == AF_INET6)
 	{
 		char		remote_port[32];
+		remote_port[0] = '\0';
```
[the remark this answers is not in the document]
Comment (haolinw) [reply src/backend/cdb/motion/ic_tcp.c:2235]: Others look good.

[the remark this answers is not in the document]
Comment (interma) [reply src/backend/cdb/motion/ic_tcp.c:2235]: thanks haolin, I will fix it.

**Comment by adam8157** [comment]
Could this be merged?

**Comment by interma** [comment]
@adam8157 Yes, it will be merged in the next week. (forget to ship this week)

**Pull request merged**